infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/RuntimeConsistencyChecker.java

```diff
@@ -0,0 +1,37 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.workspace.infrastructure.docker;
+
+import org.eclipse.che.api.core.ValidationException;
+import org.eclipse.che.api.core.model.workspace.config.Environment;
+import org.eclipse.che.api.core.model.workspace.config.MachineConfig;
+import org.eclipse.che.api.core.model.workspace.runtime.Machine;
+
+import java.util.Map;
+
+/** Checks whether runtime is consistent with its configuration. */
+class RuntimeConsistencyChecker {
+    void check(Environment environment, DockerInternalRuntime runtime) throws ValidationException {
```

Comment:
Good point, maybe it is better, but consider my thoughts:

I had a few options here:
1. Make _package-private_ method like

```java
@VisibleForTesting
void checkConsistency(Environment, DockerInternlRuntime) { ... }
```
it doesn't look right to me, the logic in this method can potentially grow which might complicate testing of this method or produce the other package-private methods which will pollute the class.

2. Make kind of current version of `ConsistencyChecker` with a static method(as you suggest).
Potentially we might need something from the container, which might change the method signature and requires changes of `DockerRuntimeContext` as well to inject properties and pass them to the static method. This looks like unnecessary coupling to me.

3. Make a class with non-static method as i did.
I like this option the most as all the logic related to consistency checking is decoupled from the context, you can easily change it in the future without touching `DockerRuntimeContext` at all.
It can also become an implementation of a strategy interface dynamically provided by container.

I am considering to mark it as `@Singleton` and inject to the context, but then container will be managing reference which might not be used at all. But this will be something similar to the behaviour you suggest and also it will reduce changes if checker needs dependencies.